### PR TITLE
Develop dual fix dual mode

### PR DESCRIPTION
**Pull request by haiwentelink**

### CODE REVIEW

**Comment by andriy-bilynskyy** [comment]
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_telink | https://github.com/telink-semi/hal_telink/commit/54c4302c9038d2cf412348400bcd039f5674da91 (develop) | https://github.com/telink-semi/hal_telink/commit/2a999d970883ce294dc1c60479b9ddc90b880bfe (develop_dual) | [telink-semi/hal_telink@54c4302c..2a999d97](https://github.com/telink-semi/hal_telink/compare/54c4302c9038d2cf412348400bcd039f5674da91..2a999d970883ce294dc1c60479b9ddc90b880bfe) |
| mcuboot | https://github.com/telink-semi/mcuboot/commit/1558e7ab0aadb4eac11f03befb5ccd3fa3f0aafe | https://github.com/telink-semi/mcuboot/commit/a07bc815e9484a8921e115f94d4f1121585494ce (develop_dual) | [telink-semi/mcuboot@1558e7ab..a07bc815](https://github.com/telink-semi/mcuboot/compare/1558e7ab0aadb4eac11f03befb5ccd3fa3f0aafe..a07bc815e9484a8921e115f94d4f1121585494ce) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

**Review by s07641069 — APPROVED** [review]

File: drivers/i2c/i2c_b9x.c [hunk 1]
```diff
@@ -53,7 +53,7 @@ static int i2c_b9x_configure(const struct device *dev, uint32_t dev_config)
 		break;
 
 	case I2C_SPEED_FAST:
-		i2c_speed = 400000U;
+		i2c_speed = 200000U;
```
Comment (liucoldstar) [inline drivers/i2c/i2c_b9x.c:57]: I2C标准定义Fast模式的400k。这个改动的需求是什么？

File: drivers/i2c/i2c_b9x.c [hunk 2]
```diff
@@ -101,7 +101,7 @@ static int i2c_b9x_transfer(const struct device *dev,
 		if (msgs[i].flags & I2C_MSG_READ) {
```
Comment (liucoldstar) [inline drivers/i2c/i2c_b9x.c:105]: read和write操作对i2c地址的定义有冲突。

File: boards/riscv/tl321x/tl3218x-common.dtsi [hunk 3]
```diff
@@ -212,7 +212,7 @@
 
 &pwm0 {
 	status = "okay";
-	clock-frequency = <300000>;
+	clock-frequency = <10000>;
```
Comment (liucoldstar) [inline boards/riscv/tl321x/tl3218x-common.dtsi]: 这个修改是基于客户需求做的吗？

File: kernel/init.c [hunk 4]
```diff
@@ -540,6 +540,17 @@ FUNC_NORETURN void z_cstart(void)
 	z_sys_init_run_level(INIT_LEVEL_PRE_KERNEL_1);
 	z_sys_init_run_level(INIT_LEVEL_PRE_KERNEL_2);
 
+#if CONFIG_STARTUP_OPTIMIZATE
```
Comment (liucoldstar) [inline kernel/init.c]: 建议改成CONFIG_STARTUP_OPTIMIZATION

File: kernel/Kconfig [hunk 5]
```diff
@@ -961,4 +961,10 @@ config HAS_DYNAMIC_DEVICE_HANDLES
 
 endmenu
 
+config STARTUP_OPTIMIZATE
```
Comment (liucoldstar) [inline kernel/Kconfig]: 建议改成下面的内容：
config STARTUP_OPTIMIZATION
    bool "Optimize startup procedure"
    default n
    help
      The option enables kernel-level optimizations to improve the efficiency of the startup process.

File: drivers/i2c/i2c_b9x.c (hunk 1 above)
Comment (haiwentelink) [reply drivers/i2c/i2c_b9x.c:57]: already use define CONFIG_I2C_LED

File: boards/riscv/tl321x/tl3218x-common.dtsi (hunk 3 above)
Comment (haiwentelink) [reply boards/riscv/tl321x/tl3218x-common.dtsi]: change it back .

File: kernel/init.c (hunk 4 above)
Comment (haiwentelink) [reply kernel/init.c]: changed.

File: kernel/Kconfig (hunk 5 above)
Comment (haiwentelink) [reply kernel/Kconfig]: changed.

File: drivers/i2c/i2c_b9x.c (hunk 2 above)
Comment (haiwentelink) [reply drivers/i2c/i2c_b9x.c:105]: use define CONFIG_I2C_LED

**Pull request merged**